en/docs/learn/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes.md

```diff
@@ -4,90 +4,209 @@ Scopes enable fine-grained access control to API resources based on user roles.
 
-1. Login to API Portal(https://localhost:9443/devportal/)
+1. Login to API Developer Portal (https://localhost:9443/devportal/).
 
-2. Navigate to the API which has the scope protected API resource and go to Credential section to subscribe to an application.
+2. Navigate to the API which has the scope protected API resource and go to **Subscription** section to subscribe to an application.
 
-    <a href="../../../../../assets/img/learn/subscribe-weatherapi.png" ><img src="../../../../../assets/img/learn/subscribe-weatherapi.png" alt="Subscribe" 
-         title="Subscribe" width="50%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/subscribe-api.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/subscribe-api.png" alt="Subscribe" 
+          title="Subscribe" width="50%" /></a>
          
-3. Generate production keys for the application.
+3. Generate PRODUCTION or SANDBOX keys for the application.
 
-    <a href="../../../../../assets/img/learn/generate-app-keys.png" ><img src="../../../../../assets/img/learn/generate-app-keys.png" alt="Generate App Keys" 
-         title="Generate App Keys" width="60%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" alt="Generate App Keys" 
+          title="Generate App Keys" width="60%" /></a>
 
-4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from the prompted window and generate the access token.
+4. Click on **GENERATE ACCESS TOKEN** button. Then select the scope from dropdown in the prompted window and click **GENERATE** button 
+to generate the access token.
 
-    <a href="../../../../../assets/img/learn/generate-access-token-with-scopes.png" ><img src="../../../../../assets/img/learn/generate-access-token-with-scopes.png" alt="Generate App Keys" 
-         title="Generate App Keys" width="50%" /></a>
+     <a href="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/generate-keys.png" ><img src="{{base_path}}/assets/img/learn/api-security/oauth2/oauth2-scopes/token-scopes.png" alt="Select Token Scopes" 
+          title="Select Token Scopes" width="60%" /></a>
 
-5. If the user has the roles specified in the scope, the access token will be issued with the requested scope. Unless, only the default scopes will be returned with the access token.
+5. If the user has the roles specified in the scope, the access token will be issued with the requested scope. Unless,
+only the default scopes will be returned with the access token.
```

Comment:
unncessary space in this line

Reply:
Unless -> Otherwise

Reply:
What are the default scopes? how to define them? where are they defined?

Reply:
its the default scope. in client credentials pov, this will also include "am_application_scope". Will add a note.  Btw fyi this is also in the 3.1 and 3.0 docs well.